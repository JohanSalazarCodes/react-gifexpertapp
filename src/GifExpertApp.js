import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({defaultCategories = []}) => {

    //const categories = ['One Punch', 'Samurai X', 'Dragon Ball'];
    const [categories, setCategories] = useState(defaultCategories);

    // const handleAdd = () => {
    //     setCategories([...categories, 'Bleach']);
    // }

    return (
        <>
            <h1>GifExpertApp</h1>
            <h1>Ivannia2</h1>
            <AddCategory setCategories = { setCategories } />
            <hr/>

           
            <ol>
                {
                    categories.map(category => (
                        <GifGrid 
                            key= {category}
                            category= { category } />
                    ))
                }
            </ol>

        </>
    );
};


