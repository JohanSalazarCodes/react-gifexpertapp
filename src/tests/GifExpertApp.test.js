import React from 'react';
import { shallow } from 'enzyme';
import  { GifExpertApp } from '../GifExpertApp';
import { GifGrid } from '../components/GifGrid';

describe('Prueba sobre componente <GifExpertApp />', () => {
   
    test('Debe mostrar el componente correctamente', () => {
       
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
        
    });

    test('debe de mostrar una lista de categorias', () => {
       const categories = ['One Punch', 'Dragon Ball'];
       const wrapper = shallow(<GifExpertApp defaultCategories={ categories } />) ;

       expect(wrapper).toMatchSnapshot();
       expect(wrapper.find('GifGrid').length).toBe(categories.length);
    });
    
    
});
