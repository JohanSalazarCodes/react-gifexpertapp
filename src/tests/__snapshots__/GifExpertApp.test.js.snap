// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba sobre componente <GifExpertApp /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Prueba sobre componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
